sourcegraph/sourcegraph.proto

```diff
@@ -588,6 +588,215 @@ service Repos {
 	rpc ListCommitters(ReposListCommittersOp) returns (CommitterList);
 }
 
+// StorageError represents an error when interacting with the Storage service.
+message StorageError {
+	// Code represents the type of error for programatic handling.
+	enum Code {
+		// NONE is the error code used when programatic handling of the error is not
+		// advised. It can be assumed that the operation failed, and that a human
+		// readable message is also provided.
+		NONE = 0;
+
+		// EOF is the error returned by Read when no more input is available.
+		// Functions should return EOF only to signal a graceful end of input. If
+		// the EOF occurs unexpectedly in a structured data stream, the appropriate
+		// error is either UNEXPECTED_EOF or some other error message giving more
+		// detail.
+		EOF = 1;
+
+		// NOT_EXIST is the error used when attempting to read/write to an object
+		// that does not exist.
+		NOT_EXIST = 2;
+
+		// PERMISSION is the error used when permission to access the given file is
+		// not granted to you.
+		PERMISSION = 3;
```

Comment:
It's actually interesting this is the case. I think we should switch to uppercase since this is what the official protobuf style guide says to do: https://developers.google.com/protocol-buffers/docs/style#enums

I also agree when translated to Go via `protoc-gen-gogo` these should be turned into CamelCase like `StorageErrorNone` `StorageErrorEOF` `StorageErrorNotExist` etc.

In the meantime, I will follow our current style as you suggest and will create an issue for this.